### PR TITLE
Add CRUD data store create and update actions

### LINKED ISSUE #33437: Create/edit product attributes via products data store API (experimental)
### Description

Add the code to be able to create and edit product attributes through the products data store.

_For now, this should be marked as experimental/internal as we iterate on the API so that it is clear that this is incomplete and subject to change_.

**Properties**
| Parameter | Type | Description
| --------- | -------- | -------- |
| `id` | integer | Unique identifier for the resource. |
| `name` | string | Attribute name. |
| `slug` | string | An alphanumeric identifier for the resource unique to its type. |
| `type` | string | Type of attribute. By default only `select` is supported. |
| `order_by` | string | Default sort order. Options: `menu_order`, `name`, `name_num` and `id`. Default is `menu_order`. |
| `has_archives` | boolean | Enable/Disable attribute archives. Default is `false`. |

#### _Create_
**HTTP Request**
> **POST** `/wp-json/wc/v3/products/attributes`

**Reference**
> https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-product-attribute

#### _Edit_
**HTTP Request**
> **PUT** `/wp-json/wc/v3/products/attributes/<id>`

**Reference**
> https://woocommerce.github.io/woocommerce-rest-api-docs/#update-a-product-attribute

### Acceptance criteria
- [ ] Data store updates
- [ ] Unit tests
- [ ] Documentation with examples

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the create and update actions to crud data store utilities.  This adds those methods to the product attributes data store which is already making use of the CRUD data store.

Closes #33437 .

### How to test the changes in this Pull Request:

1. Try to create a product attribute: `wp.data.dispatch('wc/admin/products/attributes').createProductAttribute({ name: 'Test Attribute' });`
2. Visit `/wp-admin/edit.php?post_type=product&page=product_attributes` and make sure your attribute has been created
3. Find the ID of the attribute (you can pull attributes by `wp.data.select('wc/admin/products/attributes').getProductAttributes({});`)
4. Attempt to update a product attribute via the ID `wp.data.dispatch('wc/admin/products/attributes').updateProductAttribute(3, { name: 'New Attribute Name' });`
5. Check that readme updates make sense

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33571/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Fix whitespace issues` 604e052930aaae2f8eb146a6ad938a97fc41648c
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: packages/js/data/src/crud/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+						data: {
+							...itemData,
+							[ payload.id ]: {
+								...( itemData[ payload.id ] || {} ),
+								...payload.item,
+							},
+						},
+					};
+
+                case TYPES.GET_ITEM_ERROR:
```
Comment (octaedro) [inline packages/js/data/src/crud/reducer.ts]: I think that we should add the `UPDATE_ITEM_ERROR` case here.
```
case TYPES.UPDATE_ITEM_ERROR:
```

File: packages/js/data/src/crud/test/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+		const error = 'Baaam!';
+		const state = reducer( defaultState, {
+			type: TYPES.UPDATE_ITEM_ERROR,
+			id,
+			error,
+			errorType: CRUD_ACTIONS.UPDATE_ITEM,
+		} );
+
+		expect( state.errors[ resourceName ] ).toBe( error );
+	} );
```
Comment (octaedro) [inline packages/js/data/src/crud/test/reducer.ts:191]: This test is failing but it will work after adding the line I mentioned above.

**Review by octaedro — COMMENTED** [review]
This looks great @joshuatf! This is testing well, and the code LGTM! I just left a small comment, after adding that change this PR will be ready to merge. Good job!

File: packages/js/data/src/crud/reducer.ts (hunk 1 above)
Comment (joshuatf) [reply packages/js/data/src/crud/reducer.ts]: Good catch, Fernando!  Thanks!

**Comment by joshuatf** [comment]
Thanks for the review, @octaedro!  Addressed your feedback in https://github.com/woocommerce/woocommerce/pull/33571/commits/f31103e38f1f2f0ab9afdfa209604d238ea04453.

**Review by octaedro — COMMENTED** [review]
Oh, I just noticed that there are a couple of [lint errors](https://github.com/woocommerce/woocommerce/runs/7026246227?check_suite_focus=true) in two files. Sorry for the back and forth.

**Comment by joshuatf** [comment]
Thanks @octaedro!  Fixed up those whitespace issues.

**Review by octaedro — APPROVED** [review]
Thank you @joshuatf. LGTM :shipit:

**Pull request merged**